app/src/main/java/com/matanh/transfer/server/KtorServer.kt

```diff
@@ -437,55 +442,84 @@ fun Application.ktorServer(
+                                failedUploads.add("Part processing error: ${partException.localizedMessage}")
+                            } finally {
+                                // Ensure proper cleanup of each part
+                                try {
+                                    part.dispose()
+                                } catch (disposeException: Exception) {
+                                    logger.w("Error disposing part: ${disposeException.message}")
+                                }
                             }
-                            part.dispose()
                         }
-                        if (filesUploadedCount > 0) {
-                            call.respondText(
-                                "Successfully uploaded: ${
-                                    uploadedFileNames.joinToString(
-                                        ", "
-                                    )
-                                }"
-                            )
-                        } else {
-                            call.respond(
-                                HttpStatusCode.BadRequest,
-                                "No files were uploaded or upload failed."
-                            )
+                        
+                        // Provide detailed response
+                        when {
+                            filesUploadedCount > 0 && failedUploads.isEmpty() -> {
+                                call.respondText(
+                                    "Successfully uploaded ${filesUploadedCount} file(s): ${uploadedFileNames.joinToString(", ")}"
+                                )
+                            }
+                            filesUploadedCount > 0 && failedUploads.isNotEmpty() -> {
+                                call.respondText(
+                                    "Partially successful: ${filesUploadedCount} uploaded (${uploadedFileNames.joinToString(", ")}), " +
+                                    "${failedUploads.size} failed (${failedUploads.joinToString("; ")})"
+                                )
+                            }
+                            else -> {
+                                call.respond(
+                                    HttpStatusCode.BadRequest,
+                                    ErrorResponse("Upload failed: ${failedUploads.joinToString("; ")}")
+                                )
+                            }
                         }
+                        
                     } catch (e: Exception) {
-                        logger.e("Exception during file upload")
+                        logger.e("Critical exception during large file upload: ${e.message}")
                         call.respond(
                             HttpStatusCode.InternalServerError,
-                            "Upload error: ${e.localizedMessage}"
+                            ErrorResponse("Large file upload error: ${e.localizedMessage}")
```

Comment:
`Receiving large file`,`Large file upload error`, - nothing check if the file is large. this ErrorResponse sent on any file upload error